### PR TITLE
ENH: Implement FloatingArray reductions

### PR DESCRIPTION
**Pull request by dsaxton**
Mostly copy / paste from https://github.com/pandas-dev/pandas/pull/36761

@jorisvandenbossche Would this require a whatsnew note? Seems the current v1.2.0 note is fairly high level and doesn't talk about specific methods that are supported. Should we talk about this or leave as-is?

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm @jorisvandenbossche

File: pandas/conftest.py [hunk 1]
```diff
@@ -1063,6 +1063,17 @@ def any_nullable_int_dtype(request):
     return request.param
 
 
+@pytest.fixture(params=tm.FLOAT_EA_DTYPES)
+def any_nullable_float_dtype(request):
```
Comment (simonjayhawkins) [inline pandas/conftest.py:1067]: there is already a float_ea_dtype fixture at L982

Comment (jorisvandenbossche) [reply pandas/conftest.py:1067]: Indeed, I already added one in the original PR. 

In addition, we actually also have a `dtype` fixture that is only available in the `pandas/tests/arrays/floating` directory that can be used in the test below

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good! 
At some point, we should move the implementation (and tests) to a common "NumericArray", I suppose

File: pandas/conftest.py (hunk 1 above)
Comment (dsaxton) [reply pandas/conftest.py:1067]: Ok, I can remove this in favor of the other fixture for now.

In general though do we want to try to be consistent in terms of naming / location given that there's already an `any_nullable_int_dtype` fixture in the base level `conftest.py` (maybe that means not having this latter fixture)?

**Comment by jreback** [comment]
actually if you rebase I think you can push these down to BaseMaskedArray entirely (or factor out to NumericMaskedArray if need to avoid the bool collision issues)

**Comment by jreback** [comment]
great ping on green.

**Comment by jreback** [comment]
thanks

**Pull request merged**